### PR TITLE
Add replace document button

### PR DESCRIPTION
**Pull request by jenow**
## What does this PR do ?

Add a `replace` button when updating a document since we now update it instead of replacing it to keep the original `author`, therefor it was not possible anymore to remove a field of a document.
Also added a tooltip on both button to explain what is the difference between `update` and `replace`.

## Other

Fixed a bug introduced when displaying human readable dates on dates type field.

## Preview

https://deploy-preview-520--kuzzle-backoffice-last.netlify.com/

### CODE REVIEW

File: src/components/Data/Documents/Common/CreateOrUpdate.vue [hunk 1]
```diff
@@ -92,16 +92,24 @@
...
+            <a @click.prevent="cancel" class="btn-flat waves-effect">
+              Cancel
+            </a>
+
+            <button v-if="!hideId" type="submit" class="btn primary waves-effect waves-light" :disabled="submitted">
+              <i class="fa fa-plus-circle left"></i>
+              Create
+            </button>
+
+            <button v-if="hideId" type="submit" class="btn primary waves-effect waves-light" ref="update" data-position="top" data-tooltip="Partially update a document, does not remove any attributes from a document." :disabled="submitted">
```
Comment (scottinet) [inline src/components/Data/Documents/Common/CreateOrUpdate.vue]:
```suggestion
            <button v-if="hideId" type="submit" class="btn primary waves-effect waves-light" ref="update" data-position="top" data-tooltip="Update some of a document's fields (does not remove unset attributes)." :disabled="submitted">
```

File: src/components/Data/Documents/Common/CreateOrUpdate.vue [hunk 2]
```diff
@@ -92,16 +92,24 @@
...
+              <i class="fa fa-pencil-alt left"></i>
+              Update
+            </button>
+
+            <button @click.prevent="create(true)" class="btn primary waves-effect waves-light" ref="replace" data-position="top" data-tooltip="Replace a whole document, removes attributes if they are not set." :disabled="submitted" v-if="hideId">
```
Comment (scottinet) [inline src/components/Data/Documents/Common/CreateOrUpdate.vue]:
```suggestion
            <button @click.prevent="create(true)" class="btn primary waves-effect waves-light" ref="replace" data-position="top" data-tooltip="Replace the content of a document." :disabled="submitted" v-if="hideId">
```

File: src/components/Data/Documents/Update.vue [hunk 3]
```diff
@@ -90,7 +90,11 @@ export default {
       }
 
       try {
-        await this.$kuzzle.document.update(this.index, this.collection, this.$route.params.id, document, { refresh: 'wait_for' })
+        let action = 'update'
```
Comment (Yoann-Abbes) [inline src/components/Data/Documents/Update.vue]:
```suggestion
        const action = replace ? 'replace' : 'update'
```

File: src/components/Data/Documents/Update.vue [hunk 4]
```diff
@@ -90,7 +90,11 @@ export default {
...
+        if (replace === true) {
+          action = 'replace'
+        }
```
Comment (Yoann-Abbes) [inline src/components/Data/Documents/Update.vue]: You can replace
```js
let action = 'update'
if (replace === true) {
    action = 'replace'
}
```
by
```js
const action = replace ? 'replace' : 'update'
```

**Pull request merged**